Simplified/NYPLBookCellDelegate.m

```diff
@@ -81,20 +81,31 @@ - (void)didSelectDownloadForBook:(NYPLBook *)book
 
 - (void)didSelectReadForBook:(NYPLBook *)book
 { 
-  #if defined(FEATURE_DRM_CONNECTOR)
-    // Try to prevent blank books bug
-    if ((![[NYPLADEPT sharedInstance] isUserAuthorized:[[NYPLUserAccount sharedAccount] userID]
-                                           withDevice:[[NYPLUserAccount sharedAccount] deviceID]]) &&
-        ([[NYPLUserAccount sharedAccount] hasCredentials])) {
-      [NYPLAccountSignInViewController authorizeUsingExistingCredentialsWithCompletionHandler:^{
+#if defined(FEATURE_DRM_CONNECTOR)
+  // Try to prevent blank books bug
+
+  NYPLUserAccount *user = [NYPLUserAccount sharedAccount];
+  if ([user hasCredentials]
+      && ![[NYPLADEPT sharedInstance] isUserAuthorized:[user userID]
+                                            withDevice:[user deviceID]]) {
+    // NOTE: This was cut and pasted while refactoring preexisting work:
+    // "This handles a bug that seems to occur when the user updates,
+    // where the barcode and pin are entered but according to ADEPT the device
+    // is not authorized. To be used, the account must have a barcode and pin."
+    NYPLReauthenticator *reauthenticator = [[NYPLReauthenticator alloc] init];
+    [reauthenticator authenticateIfNeeded:user
+                 usingExistingCredentials:YES
+                 authenticationCompletion:^{
+      dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Not important: Do we want to use `[NYPLMainThreadRun asyncIfNeeded:]` here just to keep things clean/consistent?

Reply:
in general I have done that, but in this case since this logic is so touchy I left it as it was before. To test this I had do manually force the code to go here because I wasn't able to reproduce the conditions naturally.